File: python-wrapper/src/python/model/embed/PersonNameUse.py
from typing import Union, Dict
from enum import Enum

class PersonNameUse(Enum):
	Usual = "usual"
	Official = "official"
	Temp = "temp"
	Nickname = "nickname"
	Anonymous = "anonymous"
	Maiden = "maiden"
	Old = "old"
	Other = "other"

	def __serialize__(self) -> object:
		return self.value

	@classmethod
	def _deserialize(cls, data: Union[str, Dict[str, object]]) -> 'PersonNameUse':
		if data == "usual":
			return PersonNameUse.Usual
		elif data == "official":
			return PersonNameUse.Official
		elif data == "temp":
			return PersonNameUse.Temp
		elif data == "nickname":
			return PersonNameUse.Nickname
		elif data == "anonymous":
			return PersonNameUse.Anonymous
		elif data == "maiden":
			return PersonNameUse.Maiden
		elif data == "old":
			return PersonNameUse.Old
		elif data == "other":
			return PersonNameUse.Other
		else:
			raise Exception(f"{data} is not a valid value for PersonNameUse enum.")

__all__ = ['PersonNameUse']

File: cardinal-sdk/src/commonMain/kotlin/com/icure/cardinal/sdk/model/embed/PersonNameUse.kt
package com.icure.cardinal.sdk.model.embed

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlin.String

@Serializable
public enum class PersonNameUse(
	internal val dtoSerialName: String,
) {
	@SerialName("usual")
	Usual("usual"),

	@SerialName("official")
	Official("official"),

	@SerialName("temp")
	Temp("temp"),

	@SerialName("nickname")
	Nickname("nickname"),

	@SerialName("anonymous")
	Anonymous("anonymous"),

	@SerialName("maiden")
	Maiden("maiden"),

	@SerialName("old")
	Old("old"),

	@SerialName("other")
	Other("other"),
}
